designcompose/src/main/java/com/android/designcompose/Layout.kt

```diff
@@ -541,81 +568,116 @@ internal inline fun DesignFrameLayout(
+    )
+        LayoutManager.setNodeSize(layoutId, rootLayoutId, myWidth, myHeight)
+    layout(myWidth, myHeight) {
+        // Place children in the parent layout
+        placeables.forEachIndexed { index, placeable ->
+            val layoutData = placeable.designLayoutData
+            if (layoutData == null) {
+                // A null layout should only happen if the child is a node that uses one of
+                // Compose's built in layouts such as Row, Column, or LazyGrid.
+                placeable.place(0, 0)
+                if (index != 0) Log.d(TAG, "Place error no layoutData: $name index $index")
+            } else {
+                val childLayout = LayoutManager.getLayoutWithDensity(layoutData.layoutId)
+                if (childLayout == null) {
+                    Log.d(TAG, "Place error null layout: $name child $index layoutId $layoutId")
                 } else {
-                    val childLayout = LayoutManager.getLayoutWithDensity(layoutData.layoutId)
-                    if (childLayout == null) {
-                        Log.d(TAG, "Place error null layout: $name child $index layoutId $layoutId")
-                    } else {
-                        placeable.place(x = childLayout.left(), y = childLayout.top())
+                    var hScrollOffset = 0
+                    var vScrollOffset = 0
+                    if (designScroll != null) {
+                        // If designScroll is not null, scroll is enabled on this node. Get the
+                        // scroll offset and offset all children
+                        if (designScroll.orientation == Orientation.Horizontal)
+                            hScrollOffset = designScroll.scrollOffset.roundToInt()
+                        else vScrollOffset = designScroll.scrollOffset.roundToInt()
+                        if (index == placeables.size - 1) {
+                            // Calculate the extents of the children by using the last item's size
+                            // and position
+                            val density = LayoutManager.getDensity()
+                            if (designScroll.orientation == Orientation.Horizontal) {
+                                val hMargin = view.style.padding.end.pointsAsDp(density).value
+                                designScroll.scrollMax.value =
+                                    (childLayout.left + childLayout.width - myWidth + hMargin)
+                                        .coerceAtLeast(0F)
+                            } else {
+                                val vMargin = view.style.padding.bottom.pointsAsDp(density).value
+                                designScroll.scrollMax.value =
+                                    (childLayout.top + childLayout.height - myHeight + vMargin)
+                                        .coerceAtLeast(0F)
+                            }
+                        }
                     }
+                    placeable.place(
```

Comment:
Compose's own scrollable implementation uses [`placeable.placeWithLayer`](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/Scroll.kt;l=396) which means that the rendering of the children will be cached in a display list (which seems like a good choice for these containers because most likely the children aren't changing much if at all).

Reply:
Done